main_window/rocket_data.py

```diff
@@ -261,8 +261,11 @@ def save(self, csv_path):
                 # Empty entry if this row of data doesn't have a val in this column
                 else:
                     data[ix, iy] = ""
-
-        np.savetxt(csv_path, np.transpose(data), delimiter=',',
+    
+        keysColumn = 0
+        sortedData = data[data[:, keysColumn].argsort()] #sort by keys in column 0
```

Comment:
I looked into it and numpy's "fancy indexing" creates a copy of the matrix. So copying the entire matrix would get slower as more rows get added over time.

I think it would be better if we did the sorting on the `keys` list higher up. Although just now I realized that that list doesn't store strings, so you'll have to also move the logic that converts keys to strings up there.